pocs/focuser/focuser.py

```diff
@@ -437,22 +437,38 @@ def _autofocus(self,
             best_focus = focus_positions[imax]
 
         elif not coarse:
-            # Crude guess at a standard deviation for focus metric, 40% of the maximum value
-            weights = np.ones(len(focus_positions)) / (spline_smoothing * metric.max())
-
-            # Fit smoothing spline to focus metric data
-            fit = UnivariateSpline(focus_positions, metric, w=weights, k=4, ext='raise')
-
-            try:
-                stationary_points = fit.derivative().roots()
-            except ValueError as err:
-                self.logger.warning('Error finding extrema of spline fit: {}'.format(err))
-                best_focus = focus_positions[imax]
-            else:
-                extrema = fit(stationary_points)
-                if len(extrema) > 0:
-                    best_focus = stationary_points[extrema.argmax()]
-                    fitted = True
+            # Fit data around the maximum value to determine best focus position.
+            # Initialise models
+            shift = models.Shift(offset=-focus_positions[imax])
+            poly = models.Polynomial1D(degree=4, c0=1, c1=0, c2=-1e-2, c3=0, c4=-1e-4,
```

Comment:
Any chance you'll want to change the degree of this, i.e. pull from config?

Reply:
It's possible, but unlikely I think. As well as changing the degree of the polynomial you'd need to change the number of points around the maximum used for the fitting, so it isn't an under or over constrained fitting problem.